src/vaults/AsyncRequests.sol

```diff
@@ -439,16 +449,36 @@ contract AsyncRequests is BaseInvestmentManager, IAsyncRequests {
         state.maxWithdraw = state.maxWithdraw > assetsUp ? state.maxWithdraw - assetsUp : 0;
 
         if (assetsDown > 0) {
-            VaultDetails memory vaultDetails = poolManager.vaultDetails(address(vault_));
-
-            if (vaultDetails.tokenId == 0) {
-                SafeTransferLib.safeTransferFrom(vaultDetails.asset, address(escrow), receiver, assetsDown);
-            } else {
-                IERC6909(vaultDetails.asset).transferFrom(address(escrow), receiver, vaultDetails.tokenId, assetsDown);
-            }
+            _withdraw(vaultAddr, receiver, assetsDown);
         }
     }
 
+    /// @dev Transfer funds from escrow to receiver and update holdings
+    function _withdraw(address vaultAddr, address receiver, uint128 assets) internal {
+        VaultDetails memory vaultDetails = poolManager.vaultDetails(vaultAddr);
+
+        IAsyncVault vault_ = IAsyncVault(vaultAddr);
+        uint64 poolId = vault_.poolId();
+        bytes16 scId = vault_.trancheId();
+        JournalEntry[] memory journalEntries = new JournalEntry[](0);
```

Comment:
Always having to initialize this is imo a strong argument for adding a `deposit` and `withdraw` function to `BalanceSheet` without the `meta` param @mustermeiszer

Reply:
We would also need new serialize/deserialize methods I guess. Is there no way in solidity to pass like a null ptr, indicating the array is empty?